deps/external-lapack.cmake

```diff
@@ -32,7 +32,7 @@ ExternalProject_Add(
   PREFIX ${CMAKE_BINARY_DIR}/lapack
   GIT_REPOSITORY ${lapack_git_repository}
   GIT_TAG ${lapack_git_tag}
-  CMAKE_ARGS ${lapack_cmake_args} -DCBLAS=ON -DLAPACKE=OFF -DBUILD_TESTING=OFF
+  CMAKE_ARGS ${lapack_cmake_args} -DCBLAS=ON -DLAPACKE=OFF -DBUILD_TESTING=OFF -DCMAKE_Fortran_FLAGS='-fno-optimize-sibling-calls'
```

Comment:
I would add a comment line saying why this flag is being added.     You can refer to the external PR https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90329 as perhaps this can be removed in the near future.

Reply:
Now that I look at this again not sure if you are getting the indended FLAGS += result.  If you see huge differences in speed this might be something to verify.

Reply:
Results look fine to me. Merging the fix in.